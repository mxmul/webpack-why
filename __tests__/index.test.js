import execa from 'execa';

describe('webpack-why', () => {
    it('responds to --help', async () => {
        const { stdout } = await execa('./lib/index.js', ['--help']);
        expect(stdout).toMatch(
            /Determine why a module is included in a Webpack bundle./,
        );
    });

    it('shows usage by default', async () => {
        await expect(execa('./lib/index.js')).rejects.toThrow(
            /Determine why a module is included in a Webpack bundle./,
        );
    });

    it('prints dependency trees', async () => {
        const { stdout } = await execa('./lib/index.js', [
            './__fixtures__/multiple-entry-points.json',
            './shared.js',
        ]);
        expect(stdout).toMatchSnapshot();
    });
});
