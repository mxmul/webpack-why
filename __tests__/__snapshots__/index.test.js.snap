// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack-why prints dependency trees 1`] = `
"Hash: 33bf293be7363a9733c7
pageB.bundle.js
└─ ./shared.js
   └─ ./pageB.js

Hash: 33bf293be7363a9733c7
pageA.bundle.js
└─ ./shared.js
   └─ ./pageA.js
"
`;
